.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: E2E Tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+jobs:
+  config:
+    env:
+      TEST_CMD: gotestsum --junitfile e2e-tests.xml --format standard-verbose --
+      UNIT_TAGS: unit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: GitHubSecurityLab/actions-permissions/monitor@v1
+        with:
+          config: ${{ vars.PERMISSIONS_CONFIG }}
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: go install gotest.tools/gotestsum@latest
+      - run: |
+          set -Eeou pipefail
+          mkdir "$XDG_CONFIG_HOME/atlascli"
+          cat <<EOF > "$XDG_CONFIG_HOME/atlascli/config.toml"
+          [e2e]
+            org_id = "5e429e7706822c6eac4d5971"
+            public_api_key = "AAUMGJXA"
+            service = "cloud"
+          EOF
+      - run: make e2e-test
+        env:
+          E2E_TAGS: atlas,config
+      - name: Test Summary
+        id: test_summary
+        uses: test-summary/action@v2.4
```

Comment:
FYI:we should use ping to a gitsha if it is not an official GH action

Reply:
why?

Reply:
https://www.paloaltonetworks.com/blog/prisma-cloud/github-actions-worm-dependencies/

We went through this process with APIX integration and the security team

Reply:
this should be CLI wide though, no?

Reply:
We don’t need to create a spreadsheet for this. Instead, we should follow the security team’s recommendations, which suggest pinning third-party GitHub Actions to a specific SHA (except for Actions provided by GitHub itself). I’m not sure if our team was already aware of this for the CLI. If not, we should start applying the same approach here.

Reply:
That is my point, let's open a jira to do it across the whole repo, not just this pr, do you mind opening that jira since you have the context?

Reply:
so I'll defer that ticket to tackle this, I'm hoping you will bring it up for discussion on our triage meeting

Reply:
https://jira.mongodb.org/browse/CLOUDP-297955